server/db-operations.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var _ = require('underscore');
+
+var Cocktail = require('../models/cocktail-model');
+var utils = require('../common/utils');
+
+var _prepForDb = function (cocktail) {
+  return {
+    name: cocktail.name,
+    url: utils.formatForUrl(cocktail.name),
+    description: cocktail.description ?
+      cocktail.description :
+      '',
+    ingredients: cocktail.ingredients ?
+      cocktail.ingredients.sort() :
+      []
+  };
+};
+
+module.exports = {
+
+  createCocktail: function (cocktailBody) {
+    var cocktail = new Cocktail(_prepForDb(cocktailBody));
+    return cocktail.save()
+      .then(function (dbRes) {
+        return _.clone(dbRes);
```

Comment:
The `_.clone` stuff gets around the object mutation stuff we were seeing last night, where the data would look fine as `dbRes` but be all messed up by the time it came in as props.